### PR TITLE
Avoid race condition in Reporter errorsBuilder

### PR DESCRIPTION
**Pull request by anetaporebska**
Fixes https://github.com/AVSystem/scex/issues/17

### CODE REVIEW

File: scex-core/src/test/scala/com/avsystem/scex/compiler/MultipleEvaluationsTest.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    new DefaultJavaScexCompiler(settings)
+  }
+
+  private val completer = compiler.getCompleter[SimpleContext[Root], Any](
+    createProfile(Nil),
+    template = false,
+    header = "",
+  )
+
+  test("Multiple evaluations of invalid expressions should always fail") {
```
Comment (sebaciv) [inline scex-core/src/test/scala/com/avsystem/scex/compiler/MultipleEvaluationsTest.scala:24]: Add test case with mixed correct and incorrect expressions

Also check corner case, when invalid expression is the last one before reaching `settings.resetAfterCount` and next expression is valid

**Review by sebaciv — CHANGES_REQUESTED** [review]

**Comment by anetaporebska** [comment]
Closing since it touches core scex functionalities and the regression risk is too high
